programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -991,6 +1153,25 @@ fn cpi_common<S: SyscallInvokeSigned>(
         .feature_set
         .is_active(&feature_set::bpf_account_data_direct_mapping::id());
 
+    // When direct mapping is on, we must update all perms at once before doing
+    // account data updates. Otherwise, updating an account might write into
+    // another account whose perms we haven't updated yet. See
+    // TEST_FORBID_LEN_UPDATE_AFTER_OWNERSHIP_CHANGE.
```

Comment:
nit: this comment is stale. it should be impossible by now, thanks to `check_account_info_pointer` and the new `>= ebpf::MM_INPUT_START` checks. when both guards are combined, there is no way to create aliasing...
that means, i think we can move this back into the original loop. however, i lean on leaving this as-is just to be safer side. so, how about updating comment at least?

also, maybe related to the ci failure: https://github.com/solana-labs/solana/pull/31986#issuecomment-1696792968